waiter/integration/waiter/autoscaling_test.clj

```diff
@@ -15,55 +15,41 @@
             [waiter.client-tools :refer :all]
             [waiter.utils :as utils]))
 
-(defn- scaling-for-service-test [testing-str waiter-url threads requests-per-thread request-fn]
+(defn- scaling-for-service-test [testing-str waiter-url target-instances concurrency-level request-fn]
   (testing testing-str
-    (let [expect-tasks-running (fn [service-id pred msg]
-                                 (loop [i 1
-                                        tasks-running 0]
-                                   (if (< i (* 5 4 requests-per-thread))
-                                     (let [tasks-running (num-tasks-running waiter-url service-id :prev-tasks-running tasks-running)]
-                                       (when (not (pred tasks-running))
-                                         (Thread/sleep 1000)
-                                         (recur (inc i) (int tasks-running))))
-                                     (is nil msg))))]
-      (let [_ (log/info (str "Making canary request..."))
-            first-request (request-fn)
-            service-id (retrieve-service-id waiter-url (:request-headers first-request))]
-        (expect-tasks-running service-id #(= % 1) "Never saw initial 1 task running")
-        (future (dorun (pmap (fn [_] (dotimes [_ requests-per-thread] (request-fn))) (range threads))))
-        (expect-tasks-running service-id #(> % (/ threads 2)) (str "Never scaled up to " (/ threads 2) " task(s)"))
-        (expect-tasks-running service-id #(= % 1) "Never scaled down to 1 task")
-        (delete-service waiter-url service-id)))))
+    (let [continue-running (atom true)
+          first-request (request-fn)
+          service-id (retrieve-service-id waiter-url (:request-headers first-request))
+          count-instances (fn [] (num-instances waiter-url service-id))]
+      (is (wait-for #(= 1 (count-instances))) "First instance never started")
+      (future (dorun (pmap (fn [_] (while @continue-running (request-fn))) (range (* target-instances concurrency-level))))) 
```

Comment:
We can use `parallelize-requests` here with the `cenceled?` function passed in:

```(parallelize-requests 
  (* target-instances concurrency-level)
  1
  request-fn
  :canceled? (fn [] (not @continue-running))
  :wait-for-tasks false)```

Reply:
Honestly I think this is much clearer.